src/Polly.Contrib.DuplicateRequestCollapser.Specs/RequestCollapserTResultSpecsBase.cs

```diff
@@ -39,8 +41,62 @@ public void Executing_concurrent_duplicate_task_not_through_CollapserPolicy_shou
             actualInvocations.Should().Be(parallelism);
 
             // All executions should have been handled the same single result instance.
-            tasks.Select(t => ((Task<ResultClass>) t).Result).Distinct().Count().Should().Be(parallelism);
+            tasks.Select(t => ((Task<ResultClass>)t).Result).Distinct().Count().Should().Be(parallelism);
         }
 
+        [Theory]
+        [ClassData(typeof(RequestCollapserTestParallelisms))]
+        public void Executing_concurrent_duplicate_faulted_task_through_CollapserPolicy_should_execute_only_once_and_return_same_single_result_instance(int parallelism)
+        {
+            ResultFactory = () => throw new Exception(_rng.Next().ToString());
+            (int actualInvocations, Task[] tasks) = Execute_parallel_delegates_through_policy_with_key_strategy(parallelism, useCollapser: true, sameKey: true);
+
+            actualInvocations.Should().Be(1);
+
+            Exception first = null;
+            try
+            {
+                _ = ((Task<ResultClass>)tasks[0]).Result;
+            }
+            catch (Exception ex)
+            {
+                first = ex.InnerException;
+            }
+
+            // All executions should have been handled the same single result instance.
+            ValidateAllTasksHaveTheSameException(tasks, first.Message);
+
+            (actualInvocations, tasks) = Execute_parallel_delegates_through_policy_with_key_strategy(parallelism, useCollapser: true, sameKey: true);
+            actualInvocations.Should().Be(2);
+
+            Exception second = null;
+            try
+            {
+                _ = ((Task<ResultClass>)tasks[0]).Result;
+            }
+            catch (Exception ex)
+            {
+                second = ex.InnerException;
+            }
+            // The result of the second batch should not be the same as the first batch.
+            second.Message.Should().NotBe(first.Message);
+            ValidateAllTasksHaveTheSameException(tasks, second.Message);
+        }
+
+        private void ValidateAllTasksHaveTheSameException(Task[] tasks, string message)
+        {
+            tasks.All(t =>
+            {
+                try
+                {
+                    _ = ((Task<ResultClass>)t).Result;
```

Comment:
>((Task<ResultClass>)t).Result; [](start = 24, length = 30)

We know this should be complete so use t.IsFaulted to check and t.Exception instead of causing the exception to get thrown.

Reply:
Good catch, forgot tasks had those properties